.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseSetParser.cs

```diff
@@ -190,20 +224,146 @@ private DateTimeResolutionResult ParserTimeEveryday(string text, DateObject refD
+                    }
+
+                }
+            }
+
+            return ret;
+        }
+
+        // parse value for 15 may of every year etc
+        private DateTimeResolutionResult ParserDayEveryweek(string text, DateObject refDate)
+        {
+            var ret = new DateTimeResolutionResult();
+
+            var ers = this.config.DateExtractor.Extract(text, refDate);
 
             if (ers.Count != 1)
             {
                 return ret;
             }
 
             var afterStr = text.Replace(ers[0].Text, string.Empty);
-            var match = this.config.EachDayRegex.Match(afterStr);
+            var timeErs = this.config.TimeExtractor.Extract(afterStr, refDate);
+            var timeErs1 = this.config.TimePeriodExtractor.Extract(afterStr, refDate);
+
+            if (timeErs.Count == 0)
+            {
+                timeErs = timeErs1;
+            }
+
+            var match = this.config.EachUnitRegex.Match(afterStr);
+            if (!match.Success)
+            {
+                match = this.config.PeriodicRegex.Match(text);
+            }
 
             if (match.Success)
             {
-                var pr = this.config.TimeParser.Parse(ers[0], DateObject.Now);
+                var pr = this.config.DateParser.Parse(ers[0], DateObject.Now);
+                var eachResult = ParseEachUnit(match.Value);
 
-                ret = SetHandler.ResolveSet(ref ret, pr.TimexStr);
+                if (timeErs.Count > 0)
+                {
+                    var timePr = this.config.TimeParser.Parse(timeErs[0], DateObject.Now);
+                    ret = SetHandler.ResolveSet(ref ret, pr.TimexStr + timePr.TimexStr);
+                    if ((config.Options & DateTimeOptions.TasksMode) != 0)
+                    {
+                        ret = TasksModeSetHandler.TasksModeResolveSet(ref ret, pr.TimexStr + timePr.TimexStr + eachResult.Timex);
+                    }
+
+                }
+                else
+                {
+                    ret = SetHandler.ResolveSet(ref ret, pr.TimexStr);
+                    if ((config.Options & DateTimeOptions.TasksMode) != 0)
+                    {
+                        ret = TasksModeSetHandler.TasksModeResolveSet(ref ret, pr.TimexStr + eachResult.Timex);
+                    }
+                }
+            }
+
+            return ret;
+        }
+
+        // parse value for input date like 19th for every month
+        private DateTimeResolutionResult ParserSingleNumbermonth(string text, DateObject refDate)
+        {
+            var ret = new DateTimeResolutionResult();
+
+            List<ExtractResult> ers = null;
+            var success = false;
+
+            // remove key words of set type from text
+            var match = config.SetEachRegex.Match(text);
+            if (match.Success)
+            {
+                var trimmedText = text.Replace(match.Value, "this ");
```

Comment:
"this " can't be hardcoded here. Also not with a whitespace. This should at least be a constant and define a method to process the value, instead of directly calling a replace here.

Reply:
replaced the following line with config specific method.